File: sdk/python/capsem/_debug.py
"""Agent-oriented diagnostics through authenticated gateway HTTP."""

from __future__ import annotations

from . import _operations as api
from . import models
from ._transport import Transport


class Debug:
    def __init__(self, transport: Transport) -> None:
        self._transport = transport

    async def panics(self, *, since: str | None = None,
                     limit: int | None = None) -> models.PanicsResponse:
        return await api.get_panics(self._transport, since=since, limit=limit)

    async def triage(self, *, vm_id: str | None = None, since: str | None = None,
                     limit: int | None = None) -> models.TriageResponse:
        return await api.get_triage(self._transport, id=vm_id, since=since, limit=limit)

File: sdk/typescript/src/debug.ts
import * as api from './operations/index.js';
import * as models from './models/index.js';
import type {DiagnosticOptions, TriageOptions} from './options.js';
import type {Transport} from './transport.js';

export class Debug {
  constructor(private readonly transport: Transport) {}
  async panics(options: DiagnosticOptions = {}): Promise<models.PanicsResponse> {
    return api.getPanics(this.transport, options, options);
  }
  async triage(options: TriageOptions = {}): Promise<models.TriageResponse> {
    return api.getTriage(this.transport, {
      since: options.since ?? null, limit: options.limit ?? null, id: options.vm_id ?? null,
    }, options);
  }
}
